lib/BootBot.js

```diff
@@ -14,20 +14,45 @@ class BootBot extends EventEmitter {
     if (!options || (options && (!options.accessToken || !options.verifyToken || !options.appSecret))) {
       throw new Error('You need to specify an accessToken, verifyToken and appSecret');
     }
+    this._options = options
     this.accessToken = options.accessToken;
     this.verifyToken = options.verifyToken;
     this.appSecret = options.appSecret;
     this.broadcastEchoes = options.broadcastEchoes || false;
     this.app = express();
     this.webhook = options.webhook || '/webhook';
     this.webhook = this.webhook.charAt(0) !== '/' ? `/${this.webhook}` : this.webhook;
-    this.app.use(bodyParser.json({ verify: this._verifyRequestSignature.bind(this) }));
     this._hearMap = [];
     this._conversations = [];
   }
-
+  setBeforeMiddlewares( beforeMiddlewares ){
+    if( !( beforeMiddlewares instanceof Array ) ){
+      throw new Error( 'Paremeter in setBeforeMiddlewares() must be an array of middlewares' )
+    }else if( !beforeMiddlewares || !beforeMiddlewares.length ){
+      return
+    }
+    beforeMiddlewares.forEach( middleware => {
+      this.app.use( middleware )
+    })
+  }
+  setAfterMiddlewares( afterMiddlewares ){
+    if( !( afterMiddlewares instanceof Array ) ){
+      throw new Error( 'Paremeter in setAfterMiddlewares() must be an array of middlewares' )
+    }else if( !afterMiddlewares || !afterMiddlewares.length ){
+      return
+    }
+    afterMiddlewares.forEach( middleware => {
+      this.app.use( middleware )
+    })
+  }
   start(port) {
+    // set middlewares
+    this.setBeforeMiddlewares( this._options.beforeMiddlewares || [] )
+    this.app.use(bodyParser.json({ verify: this._verifyRequestSignature.bind(this) }));
```

Comment:
This gives us an opportunity to make this optional (but defaulting to active) in case you are handling other kinds of non-bot http requests with this express instance.
I, for example, have had to patch this in my own multipage bot in the past.

So we could make `_verifyRequestSignature` "public", add a boolean `this.shouldVerifyRequest` or something, set that default to `true` and also add it before `setBeforeMiddlewares` so we can halt the request before hitting other middlewares so we don't waste cpu cycles.

Or do you have a reason for setting it after `setBeforeMiddlewares` @JacobGadawski?